packages/core/src/tracing/instrumentation/messaging/rdkafka.js

```diff
@@ -432,36 +440,42 @@ function addTraceLevelSuppressionString(headers) {
   if (headers == null) {
     headers = [
       {
-        [constants.kafkaTraceLevelHeaderNameString]: '0'
+        [constants.kafkaTraceLevelHeaderName]: '0'
       }
     ];
   } else if (headers && Array.isArray(headers)) {
-    headers.push({ [constants.kafkaTraceLevelHeaderNameString]: '0' });
+    headers.push({ [constants.kafkaTraceLevelHeaderName]: '0' });
   }
   return headers;
 }
 
 function findInstanaHeaderValues(instanaHeadersAsObject) {
   let traceId;
+  let longTraceId;
   let parentSpanId;
   let level;
 
   // CASE: Look for the the newer string header format first.
-  if (instanaHeadersAsObject[constants.kafkaTraceIdHeaderNameString]) {
-    traceId = String(instanaHeadersAsObject[constants.kafkaTraceIdHeaderNameString]);
+  if (instanaHeadersAsObject[constants.kafkaTraceIdHeaderName]) {
+    traceId = String(instanaHeadersAsObject[constants.kafkaTraceIdHeaderName]);
+    if (traceId) {
+      const limited = limitTraceId({ traceId });
```

Comment:
Note: Limiting the incoming trace ID here to 16 characters is one crucial change of this PR.